File: java/lib/src/main/java/com/svix/api/DestinationTransformation.java
// this file is @generated
package com.svix.api;

import com.svix.SvixHttpClient;
import com.svix.exceptions.ApiException;
import com.svix.models.DestinationTransformIn;
import com.svix.models.DestinationTransformationOut;
import com.svix.models.EmptyResponse;

import okhttp3.HttpUrl;

import java.io.IOException;

public class DestinationTransformation {
    private final SvixHttpClient client;

    public DestinationTransformation(SvixHttpClient client) {
        this.client = client;
    }

    /** Get the transformation code associated with this destination. */
    public DestinationTransformationOut get(final String appId, final String destinationId)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(
                                String.format(
                                        "/api/v1/app/%s/destination/%s/transformation",
                                        appId, destinationId));
        return this.client.executeRequest(
                "GET", url.build(), null, null, DestinationTransformationOut.class);
    }

    /** Set or unset the transformation code associated with this destination. */
    public EmptyResponse patch(
            final String appId,
            final String destinationId,
            final DestinationTransformIn destinationTransformIn)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(
                                String.format(
                                        "/api/v1/app/%s/destination/%s/transformation",
                                        appId, destinationId));
        return this.client.executeRequest(
                "PATCH", url.build(), null, destinationTransformIn, EmptyResponse.class);
    }
}

File: kotlin/lib/src/main/kotlin/DestinationTransformation.kt
// this file is @generated
package com.svix.kotlin

import com.svix.kotlin.models.DestinationTransformIn
import com.svix.kotlin.models.DestinationTransformationOut
import com.svix.kotlin.models.EmptyResponse

class DestinationTransformation(private val client: SvixHttpClient) {
    /** Get the transformation code associated with this destination. */
    suspend fun get(appId: String, destinationId: String): DestinationTransformationOut {
        val url =
            client
                .newUrlBuilder()
                .encodedPath("/api/v1/app/$appId/destination/$destinationId/transformation")
        return client.executeRequest<Any, DestinationTransformationOut>("GET", url.build())
    }

    /** Set or unset the transformation code associated with this destination. */
    suspend fun patch(
        appId: String,
        destinationId: String,
        destinationTransformIn: DestinationTransformIn,
    ): EmptyResponse {
        val url =
            client
                .newUrlBuilder()
                .encodedPath("/api/v1/app/$appId/destination/$destinationId/transformation")

        return client.executeRequest<DestinationTransformIn, EmptyResponse>(
            "PATCH",
            url.build(),
            reqBody = destinationTransformIn,
        )
    }
}
